run_model.py

```diff
@@ -0,0 +1,126 @@
+# app.py
+from flask import Flask, request, jsonify
+from transformers import AutoTokenizer, AutoModelForCausalLM
+import requests
+import torch
+import os
+
+app = Flask(__name__)
+
+MODEL_DIR = './models'
+
+os.makedirs(MODEL_DIR, exist_ok=True)
+
+# Define globals
+MODEL = None
+TOKEN = None
+TOKENIZER = None
+DEVICE = None
+MODEL_NAME = None
+
+def validate_huggingface_token(token: str) -> bool:
+    headers = {"Authorization": f"Bearer {token}"}
+    response = requests.get("https://huggingface.co/api/whoami-v2", headers=headers)
+    return response.status_code == 200
+
+def set_model_name(model_name: str):
+    global MODEL, TOKEN, TOKENIZER, DEVICE, MODEL_NAME  # Reference the globals
+    response = create_model(model_name, TOKEN)
+    if response is not None:
+        MODEL, TOKENIZER, DEVICE = response
+        MODEL_NAME = model_name
+        return True
+    else:
+        return False
+    
+def set_token(new_token:str):
+    global MODEL, TOKEN, TOKENIZER, DEVICE, MODEL_NAME  # Reference the globals   
+    validate = validate_huggingface_token(new_token)
+    response = create_model(MODEL_NAME, new_token)
+    if validate and response is not None:
+        MODEL, TOKENIZER, DEVICE = response
+        TOKEN = new_token
+        return True
+    else:
+        return False
+
+
+def create_model(model_name: str, token: str):
+    try:
+        # Check if the model is already in the model directory, otherwise download it
+        if not os.path.exists(os.path.join(MODEL_DIR, model_name)):
+            tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True, cache_dir=MODEL_DIR, token=token)
+            # Add a new [PAD] token
+            tokenizer.add_special_tokens({'pad_token': '[PAD]'})    
+            model = AutoModelForCausalLM.from_pretrained(model_name, trust_remote_code=True, cache_dir=MODEL_DIR, token=token, pad_token_id=tokenizer.eos_token_id)
+        else:
+            tokenizer = AutoTokenizer.from_pretrained(MODEL_DIR, trust_remote_code=True, token=token)
+            # Add a new [PAD] token
+            tokenizer.add_special_tokens({'pad_token': '[PAD]'})
+            model = AutoModelForCausalLM.from_pretrained(MODEL_DIR, trust_remote_code=True, token=token, pad_token_id=tokenizer.eos_token)
+        
+        # Move model to GPU if available
+        device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+        model.to(device)
+        return model, tokenizer, device
+    except Exception as e:
+        return None
+    
+def init_model():
+    global MODEL, TOKEN, MODEL_NAME  # Reference the globals
+    # Model details
+    # meta-llama/Llama-3.2-1B-Instruct
+    # Qwen/Qwen2.5-Coder-1.5B-Instruct
+
+    MODEL_NAME = "meta-llama/Llama-3.2-1B-Instruct"
+    TOKEN = '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'
+
+    # TOKEN1 = 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
+    # TOKEN2 = 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
```

Comment:
ça on l'enlève ?

Reply:
C'était pour tester le changement de token, j'ai noté 2 token valide là mais on peut l'enlever